src/client.js

```diff
@@ -226,6 +226,25 @@ export class Saturn {
     return { res, controller, log }
   }
 
+  async _setStorage () {
+    const getMemoryStorage = async () => {
+      const memoryStorageFn = memoryStorage()
+      return await memoryStorageFn()
+    }
+
+    if (!this.config.storage) {
+      this.storage = await getMemoryStorage()
+      return
+    }
+
+    try {
+      const getStorage = this.config.storage
+      this.storage = await getStorage()
+    } catch (error) {
+      this.storage = await getMemoryStorage()
+    }
+  }
+
```

Comment:
Hmmm.. I think the memoryStorage logic could've been left as is. What we decided on was "a function that returns a storage object", and the existing memoryStorage() usage already did that.

```suggestion
  async _setStorage () {
    if (!this.config.storage) {
      this.storage = memoryStorage()
      return
    }

    try {
      const getStorage = this.config.storage
      this.storage = await getStorage()
    } catch (error) {
      this.storage = memoryStorage()
    }
  }

```

Reply:
My take was that we were designing a factory function => A function that returns another function. 

In any case, I think all the storage interfaces should be identical. Someone could just import the `memoryStorage` and pass it as the storage dependency.

Reply:
Factory functions are just functions that return objects/instances. You use them instead of constructors when you need to return different instance types. That's the way I understand them at least.